### ISSUE TITLE
Original error: unknown error: Chrome version must be >= 53.0.2785.0

### ISSUE THREAD

**Issue by pisarenco92** [appium/appium#7138]
## The problem

I'm testing a hybrid app, when context is changing to webview an error occured: Original error: unknown error: Chrome version must be >= 53.0.2785.0 

## Environment

* Appium version (or git revision) that exhibits the issue: 1.6
* Desktop OS/version used to run Appium: macOS Sierra 10.12
* Node.js version (unless using Appium.app|exe):
* Mobile platform/version under test: Android 6.0
* Real device or emulator/simulator:simulator

## Details

I have tested with different chromedriver versions(2.21, 2.22, 2.23, 2.24, 2.25) same error. 
If I run on Android 5.1.1 no issues.

## Link to Appium logs

https://gist.github.com/pisarenco92/18061c8756cbae25b5ee968d1210560c

**Comment by Dave-Osborne** [comment]
I'm having the same problem using the current 1.6 beta 3 version of Appium when running against Android 7.1.1

**Comment by imurchie** [comment]
What version of Chrome is on the device?

**Comment by tomliuxin** [comment]
@imurchie 
how to upgrade the ChromeDriver to version 53.0.2785.0
I fond the available latest version is 2.25.1

C:\Users\v-tomliu>npm install chromedriver@53.0.2785.0
npm ERR! Windows_NT 10.0.14393
npm ERR! argv "C:\\Program Files\\nodejs\\node.exe" "C:\\Users\\v-tomliu\\AppData\\Roaming\\npm\\node_modules\\npm\\bin\\npm-cli.js" "install" "chromedriver@53.0.2785.0"
npm ERR! node v6.9.1
npm ERR! npm  v3.10.1

npm ERR! No compatible version found: chromedriver@53.0.2785.0
npm ERR! Valid install targets:
npm ERR! 2.25.1, 2.25.0, 2.24.1, 2.24.0, 2.23.1, 2.23.0, 2.22.2, 2.22.1, 2.22.0, 2.21.2, 2.21.1, 2.21.0, 2.20.0, 2.19.0, 2.18.0, 2.16.0, 2.15.2, 2.15.1, 2.15.0, 2.14.1, 2.14.0, 2.13.0, 2.12.0, 2.11.0, 2.10.0, 2.10.0-1, 2.9.1-1, 2.9.1-0, 2.9.0, 2.8.1, 2.8.0, 2.6.0, 2.3.0, 2.2.0, 2.0.1, 2.0.0

**Comment by qyfeiforfree** [comment]
me too，hope fix it as soon

**Comment by enzoliao** [comment]
I think this problem may be related to the webview version of the hybrid app. Browser in Android App is not as new as the latest Chrome.

**Comment by Dave-Osborne** [comment]
@imurchie My emulator is running 53.0.2785.134

**Comment by tomliuxin** [comment]
@Dave-Osborne  how are you upgrade the chrome to 53.0.2785.134?

**Comment by Dave-Osborne** [comment]
@tomliuxin it seems to be the version which is included in Android 7.1.1 (or it updated itself).

I'm running Appium from source, which references all of the current dependencies.
https://github.com/appium/appium/blob/master/docs/en/contributing-to-appium/appium-from-source.md

**Comment by tomliuxin** [comment]
@Dave-Osborne , Yes, I am running at the Android 7.0, the chrome version is 51.***, due to it installed when emulator created, so I can't uninstall/re-install it. 

I will try to use Android 7.1.1. Thanks.

**Comment by tomliuxin** [comment]
@Dave-Osborne , @imurchie 
Yes,  run cases at emulator android 7.1.1 and chrome version is 53.0.2785.134, met the same as issue.

log:
https://gist.github.com/tomliuxin/6db770ac2fdceb1ca9e692f2a266076e

**Comment by kprpalash82** [comment]
Getting the same issue in Hybrid App for Android 6.0 API 23, Appium 1.6, ChromeDriver 2.25. But when tried with  Android 7.0 API and ChromeDriver 2.23 it worked.

**Comment by imurchie** [comment]
And Android 6.0 with ChromeDriver 2.23? 

It is a juggling game. CD requires a minimum Chrome version with each release. You can move back to earlier versions of CD to make it work.

**Comment by kprpalash82** [comment]
Rightly said @imurchie it is indeed a juggling game, for Android 6.0 CD 2.20 and 2.19 are working perfectly fine.

**Comment by pisarenco92** [comment]
Thanks @imurchie @kprpalash82 Changed chrome driver to 2.20 and everything is working fine.

**Comment by tomliuxin** [comment]
@pisarenco92 , how to upgrade the chrome driver to 2.20.0?

I met the below issue when upgrade: **npm -g install chromedriver@2.20.0**

C:\Users\v-tomliu>npm -g install chromedriver@2.20.0
npm ERR! fetch failed https://registry.npmjs.org/chromedriver/-/chromedriver-2.20.0.tgz
npm WARN retry will retry, error on last attempt: Error: fetch failed with status code 503
npm ERR! fetch failed https://registry.npmjs.org/chromedriver/-/chromedriver-2.20.0.tgz
npm WARN retry will retry, error on last attempt: Error: fetch failed with status code 503
npm ERR! fetch failed https://registry.npmjs.org/chromedriver/-/chromedriver-2.20.0.tgz
npm ERR! Windows_NT 10.0.14393
npm ERR! argv "C:\\Program Files\\nodejs\\node.exe" "C:\\Users\\v-tomliu\\AppData\\Roaming\\npm\\node_modules\\npm\\bin\\npm-cli.js" "-g" "install" "chromedriver@2.20"
npm ERR! node v6.9.1
npm ERR! npm  v3.10.1

npm ERR! fetch failed with status code 503
npm ERR!
npm ERR! If you need help, you may report this error at:
npm ERR!     <https://github.com/npm/npm/issues>

npm ERR! Please include the following file with any support request:
npm ERR!     C:\Users\v-tomliu\npm-debug.log

**Comment by kprpalash82** [comment]
@tomliuxin : Instead of upgrading i would suggest download latest chromedriver from https://chromedriver.storage.googleapis.com/index.html and pass the path of driver to appium server as a argument. Something like : appium -a 10.171.50.94 -p 4723 --app /Users/abc/Downloads/someapp.apk --chromedriver-executable /Users/abc/Downloads/ChromeDrivers/20/chromedriver

**Comment by tomliuxin** [comment]
@kprpalash82 , Thanks. 
worked fine for 2.20.0 CD version, NO met this issue. 

I am using the C# code, after change the driver context from NATIVE_APP to WEBVIEW_***, and try to use the below code to get the screen size, met the issue:
 int Screen_Y = myDriver.Manage().Window.Size.Height

the issue:
[debug] [Chromedriver] Changed state to 'online'
[MJSONWP] Responding to client with driver.setContext() result: null
[HTTP] <-- POST /wd/hub/session/2bd9612e-5f5c-42d9-907e-8ed7dd3ca284/context 200 8285 ms - 76
[HTTP] --> GET /wd/hub/session/2bd9612e-5f5c-42d9-907e-8ed7dd3ca284/window/current/size {}
[MJSONWP] Driver proxy active, passing request on via HTTP proxy
[JSONWP Proxy] Proxying [GET /wd/hub/session/2bd9612e-5f5c-42d9-907e-8ed7dd3ca284/window/current/size] to [GET http://127.0.0.1:8000/wd/hub/session/f0e1c69aeddffb14d868f9c8ba205037/window/current/size] with body: {}
[JSONWP Proxy] Got response with status 200: "{\"sessionId\":\"f0e1c69aeddffb14d868f9c8ba205037\",\"status\":13,\"value\":{\"message\":\"unknown error: operation is unsupported on Android\\n  (Session info: chrome=51.0.2704.91)\\n  (Driver info: chromedriver=2.20.353145 (343b531d31eeb933ec778dbcf7081628a1396067),platform=Windows NT 10.0 x86_64)\"}}"

**Comment by ntulele** [comment]
Just want to reiterate that downgrading chromedriver to 2.20 solved the issue for me.
_My Environment:_
AVD 6.0
Appium 1.5.3 
java-client 3.4.1

**Comment by ltiuaopaul** [comment]
When I changed chrome driver to 2.20 or 2.19, sometimes it working fine but need to wait for a long time,  sometimes an error occurred("[Chromedriver] Error: chrome not reachable"). It seems that switch_to_webview is not stable,  is there any solution for this issue?
My Environment:
Win7+android6.0+Appium1.6+python
Error log:
2016-11-10 01:34:04:172 - error: [Chromedriver] Error: chrome not reachable
  (Driver info: chromedriver=2.19.346078 (6f1f0cde889532d48ce8242342d0b84f94b114a1),platform=Windows NT 6.1 SP1 x86_64)
    at Chromedriver.callee$2$0$ (lib/chromedriver.js:177:15)
    at tryCatch (C:\Users\administrator\AppData\Roaming\npm\node_modules\appium\node_modules\appium-android-driver\node_modules\appium-chromedriver\node_modules\babel-runtime\regenerator\runtime.js:67:40)
    at GeneratorFunctionPrototype.invoke [as _invoke] (C:\Users\administrator\AppData\Roaming\npm\node_modules\appium\node_modules\appium-android-driver\node_modules\appium-chromedriver\node_modules\babel-runtime\regenerator\runtime.js:315:22)
    at GeneratorFunctionPrototype.prototype.(anonymous function) [as next] (C:\Users\administrator\AppData\Roaming\npm\node_modules\appium\node_modules\appium-android-driver\node_modules\appium-chromedriver\node_modules\babel-runtime\regenerator\runtime.js:100:21)
    at GeneratorFunctionPrototype.invoke (C:\Users\administrator\AppData\Roaming\npm\node_modules\appium\node_modules\appium-android-driver\node_modules\appium-chromedriver\node_modules\babel-runtime\regenerator\runtime.js:136:37)
 Error: chrome not reachable
  (Driver info: chromedriver=2.19.346078 (6f1f0cde889532d48ce8242342d0b84f94b114a1),platform=Windows NT 6.1 SP1 x86_64)
    at Chromedriver.callee$2$0$ (lib/chromedriver.js:177:15)
    at tryCatch (C:\Users\administrator\AppData\Roaming\npm\node_modules\appium\node_modules\appium-android-driver\node_modules\appium-chromedriver\node_modules\babel-runtime\regenerator\runtime.js:67:40)
    at GeneratorFunctionPrototype.invoke [as _invoke] (C:\Users\administrator\AppData\Roaming\npm\node_modules\appium\node_modules\appium-android-driver\node_modules\appium-chromedriver\node_modules\babel-runtime\regenerator\runtime.js:315:22)
    at GeneratorFunctionPrototype.prototype.(anonymous function) [as next] (C:\Users\administrator\AppData\Roaming\npm\node_modules\appium\node_modules\appium-android-driver\node_modules\appium-chromedriver\node_modules\babel-runtime\regenerator\runtime.js:100:21)
    at GeneratorFunctionPrototype.invoke (C:\Users\administrator\AppData\Roaming\npm\node_modules\appium\node_modules\appium-android-driver\node_modules\appium-chromedriver\node_modules\babel-runtime\regenerator\runtime.js:136:37)
2016-11-10 01:34:04:173 - error: [MJSONWP] Encountered internal error running command: Error: chrome not reachable
  (Driver info: chromedriver=2.19.346078 (6f1f0cde889532d48ce8242342d0b84f94b114a1),platform=Windows NT 6.1 SP1 x86_64)
    at Chromedriver.callee$2$0$ (lib/chromedriver.js:177:15)
    at tryCatch (C:\Users\administrator\AppData\Roaming\npm\node_modules\appium\node_modules\appium-android-driver\node_modules\appium-chromedriver\node_modules\babel-runtime\regenerator\runtime.js:67:40)
    at GeneratorFunctionPrototype.invoke [as _invoke] (C:\Users\administrator\AppData\Roaming\npm\node_modules\appium\node_modules\appium-android-driver\node_modules\appium-chromedriver\node_modules\babel-runtime\regenerator\runtime.js:315:22)
    at GeneratorFunctionPrototype.prototype.(anonymous function) [as next] (C:\Users\administrator\AppData\Roaming\npm\node_modules\appium\node_modules\appium-android-driver\node_modules\appium-chromedriver\node_modules\babel-runtime\regenerator\runtime.js:100:21)
    at GeneratorFunctionPrototype.invoke (C:\Users\administrator\AppData\Roaming\npm\node_modules\appium\node_modules\appium-android-driver\node_modules\appium-chromedriver\node_modules\babel-runtime\regenerator\runtime.js:136:37)

**Comment by hardik-d** [comment]
@imurchie i am testing a hybrid app on android 7 emulator and i am seeing:
Selenium::WebDriver::Error::TimeOutError:
       timed out after 15 seconds (An unknown server-side error occurred while processing the command. Original error: unknown error: Chrome version must be >= 53.0.2785.0
         (Driver info: chromedriver=2.25.426935 (820a95b0b81d33e42712f9198c215f703412e1a1),platform=Mac OS X 10.11.6 x86_64))

I am on appium 1.6.1 beta.

Is there a documentation mapping what chromedriver version to have for which android version or are you planning to downgrade it @imurchie

**Comment by imurchie** [comment]
There is no plan to downgrade it, no. But you can specify any chromedriver binary you want, by specifying the path to it through the `--chromedriver-executable` server argument. Appium tries to keep up with the latest version, so as to include upstream fixes as much as possible.

Downloads can be found here: https://sites.google.com/a/chromium.org/chromedriver/downloads

What version of Chrome is on the device?

**Comment by hardik-d** [comment]
chrome browser version is 53

**Comment by imurchie** [comment]
Looks like CD 2.24 ought to handle that. https://chromium.googlesource.com/chromium/src/+/117477d1e4408e614a2a8c90c5b3a0a88626841d/chrome/test/chromedriver/chrome/version.cc

**Comment by gustavomick** [comment]
appium --chromedriver-executable pathdriver/chromedriver-2.24.exe 

worked for me

**Comment by hardik-d** [comment]
yeah but my tests are hanging after running maybe couple test @imurchie

**Comment by mayukataoka** [comment]
Somehow, this is still not working for me and I am wondering if I am doing something wrong... 

1. Downloaded chromedriver 2.24.  Unzipped the chromedriver zip file for Mac.  
2. Started appium as 
`appium  --chromedriver-executable /path-to/chromedriver  `
3. Ran my python Appium tests.

=>The error message I am still getting is:

> E       selenium.common.exceptions.WebDriverException: Message: An unknown server-side error occurred while processing the command. Original error: unknown error: Chrome version must be >= 52.0.2743.0
E         (Driver info: chromedriver=2.24.417412 (ac882d3ce7c0d99292439bf3405780058fcca0a6),platform=Mac OS X 10.12.2 x86_64)

**Comment by crclayton** [comment]
This is unrelated to this repository, but merely for future Google-searchers. I was getting this error with the latest build of ChromeDriver, 2.27. 

> selenium.common.exceptions.WebDriverException: Message: unknown error: Chrome version must be >= 54.0.2840.0
  
> (Driver info: chromedriver=2.27.440196 (3b42f4145a4fd223b41f39bb38aaf90508be7a96),platform=Linux 3.13.0-37-generic x86_64)

As mentioned by @ntulele, downgrading to version 2.20 fixed the issue.

**Comment by mmusil** [comment]
I'm testing hybrid app on android 7.1.1 with appium v1.6.4 and I'm getting similar error. I've updated Chrome in my Android Emulator to version 57.0.2987.132 but test are still failing.

`Error response status: 13, UnknownError - An unknown server-side error occurred while processing the command. Selenium error: An unknown server-side error occurred while processing the command. Original error: unknown error: Chrome version must be >= 55.0.2883.0
  (Driver info: chromedriver=2.28.455506 (18f6627e265f442aeec9b6661a49fe819aeeea1f),platform=Linux 4.9.11-200.fc25.x86_64 x86_64)
`
Have to note, that **with Android 7.0 I don't have any problems.**

**Comment by hardik-d** [comment]
install android system webview from playstore

**Comment by mmusil** [comment]
@hardik-d Didin't help. I've tried to use chromedrive 2.20 and it works for android 7.1.1, but then it's not working for Android 7.0. For 7.0 I have to switch back to the newer version :D

**Comment by gwynjudd** [comment]
Hello,

I have also come across this issue. I found something interesting about the AVD emulators in Android studio 2.3 (not sure about 2.4 yet, I'm just updating).

The summary of what I found is that in the Android 7.1.1 AVD emulator in Android Studio 2.3, the Webview Implementation does not use the installed version of chrome. Note that normally in Android 7.x, the Android WebView component is using the same version as the installed chrome. 

Here is a screenshot from the developer options in the 7.1.1 emulator, showing that the WebView implementation is using "Android WebView". Normally in 7.x, it would show options such as "Chrome Beta" or "Chrome Stable", depending on what versions of chrome you had installed.

![image](https://cloud.githubusercontent.com/assets/314188/25467444/53d586b6-2b62-11e7-8052-0db7f171782a.png)

Here is the chrome version on the emulator (chrome 53.xx):

![image](https://cloud.githubusercontent.com/assets/314188/25467518/db36a6da-2b62-11e7-947f-dcde5d0530ae.png)

Here is `navigator.appVersion` in chrome on the emulator as well (chrome 53.xx):

![image](https://cloud.githubusercontent.com/assets/314188/25467548/05c6cc72-2b63-11e7-99f1-0437b215c350.png)

Here is `navigator.appVersion` in my hybrid app (chrome 52.xx):

![image](https://cloud.githubusercontent.com/assets/314188/25467573/3369b73e-2b63-11e7-8c1e-65ccdcd82912.png)
